src/Microsoft.AspNetCore.Routing.Abstractions/Endpoint.cs

```diff
@@ -1,27 +1,35 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Http;
+using System.Threading.Tasks;
+
 namespace Microsoft.AspNetCore.Routing
 {
     /// <summary>
     /// Respresents a logical endpoint in an application.
     /// </summary>
     public abstract class Endpoint
     {
+        internal static readonly RequestDelegate EmptyInvoker = (context) => Task.CompletedTask;
+
         /// <summary>
         /// Creates a new instance of <see cref="Endpoint"/>.
         /// </summary>
+        /// <param name="invoker">The delegate used to processes requests for the endpoint.</param>
         /// <param name="metadata">
         /// The endpoint <see cref="EndpointMetadataCollection"/>. May be null.
         /// </param>
         /// <param name="displayName">
         /// The informational display name of the endpoint. May be null.
         /// </param>
         protected Endpoint(
+            RequestDelegate invoker,
```

Comment:
I think @davidfowl wanted this renamed to `requestDelegate`. I'm kinda neutral on it, as long as you guys agree I'm happy 👍